firebase-firestore/src/main/java/com/google/firebase/firestore/util/MoreObjects.java

```diff
@@ -0,0 +1,358 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore.util;
+
+import static com.google.firebase.firestore.util.Preconditions.checkNotNull;
+
+import com.google.errorprone.annotations.CanIgnoreReturnValue;
+import java.util.Arrays;
+import javax.annotation.Nullable;
+
+/**
+ * Helper functions that operate on any {@code Object}, and are not already provided in {@link
+ * java.util.Objects}.
+ *
+ * <p>See the Guava User Guide on <a
+ * href="https://github.com/google/guava/wiki/CommonObjectUtilitiesExplained">writing {@code Object}
+ * methods with {@code MoreObjects}</a>.
+ *
+ * @author Laurence Gonsalves
+ * @since 18.0 (since 2.0 as {@code Objects})
+ */
+public final class MoreObjects {
```

Comment:
Please replace the only usage of this class with a custom toString() implementation (as we have done in almost all other places of the code base).

Reply:
Done.